### ISSUE TITLE
[BUG] App freezes filling billing address with validation errors

### ISSUE THREAD

**Issue by mcarter6061** [Adyen/adyen-ios#662]
**Describe the bug**
The host app may freeze when filling in the full billing address in the Drop-In component after showing validation errors in some address fields.  Once this the validation errors show, you cannot dismiss the drop-in component and the app is permanently frozen/hung.  This occurs intermittently.

**To Reproduce**
Reproduced in the SwiftUI Demo app.  
- Change the `configuration.card.billingAddressMode` to `.full`.
- Execute the demo app on a device ( this issue does not occur in the simulator ).  My test device was an iPhone 12 Pro running iOS 14.8
- Select Drop-In
- Fill in card details with valid card, expiry, cvc
- Select UK country
- Fill in house number and street number
- Leave City and Postal code empty
- Tap Pay button
- City and Postal code fields are marked red with validation errors, and the app may now be frozen

**Expected behavior**
After tapping pay we expected to see validation errors, but the app not to have frozen.

**Screenshots**

https://user-images.githubusercontent.com/7438648/136587777-ed2b154a-0de7-43bf-a88b-c5fd3c6a4324.mov

**Smartphone (please complete the following information):**
 - Device: iPhone 12 Pro
 - OS: iOS 14.8
 - SDK Version 4.1.0

**Comment by descorp** [comment]
Hey @mcarter6061 

Thanks for feedback!
We will investigate

**Comment by descorp** [comment]
Hey @mcarter6061 

unfortunately, I wasn't able to reproduce this 🤔

Are you using device or simulator?
Is it release or debug?
Does it always happen? 
What do you see in Xcode, logs when this happen?

> Leave City and Postal code empty

Is it specific to those fields? What if you leave more / less fields empty?

**Comment by mcarter6061** [comment]
I used the latest develop on the day I posted the link (d5ac06ebde3613942ec2b5cd64e4e3496eda982d)
This was using the SwiftUI host application (**AdyenSwiftUIHost**)
I cannot reproduce on a simulator, it is device only.

It is fairly reproducible for me, so I didn't experiment with leaving other fields blank.  I just tried with another country,  Netherlands, and validation errors did not result in the app hanging.  So it seems like it could be UK specific.

A bit more experimentation:

- If you manage to get validation errors visible without hanging the app, then future taps of the Pay button do not hang the app.  Even after following the specific steps that normally reproduce it.
- I can reliably reproduce it, even after seeing validation errors by changing country away from the UK, then back to the UK, which clears all the validation errors.  Enter any value for house number, then with the keyboard visible, scroll down to the pay button and tap Pay.  At that point the validation errors will show on the other address fields and the app hangs.  
- It does not hang if I manage to dismiss the keyboard before hitting pay.  ie, fill in house number, tap country but don't change the country just tap "Done".  Now the keyboard dismisses, and then tap Pay.  The same validation errors show as above, but the app does not hang.  I would guess it has something to do with the keyboard dismissing.  
- Shortest path to reproduce (on iPhone 12 Pro (14.8)):  Select Drop-In, ignore credit card details, Tap Country and select "United Kingdom",  Select House number and type at least one character, scroll down to the Pay button and not dismiss keyboard (focus is still on house number), select "Pay".  App should hang.
- Tested on an iPhone SE (14.7.1) - cannot reproduce it.
